### PR TITLE
add script to determine PR target branch in GitLab CI

### PR DESCRIPTION
**Pull request by SimeonEhrig**

### CODE REVIEW

File: .ci/integTestGen/src/get_target_branch.jl [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    if (!startswith(splited_commit_ref_name[1], "pr-"))
+        error("CI_COMMIT_REF_NAME does not start with pr-")
+    end
+
+    # parse to Int only to check if it is a number
+    pr_number = parse(Int, splited_commit_ref_name[1][(length("pr-") + 1):end])
+    if (pr_number <= 0)
+        error(
+            "a PR number always needs to be a positiv integer number bigger than 0: $pr_number",
```
Comment (AntonReinhard) [inline .ci/integTestGen/src/get_target_branch.jl:33]:
```suggestion
            "a PR number always needs to be a positive integer number bigger than 0: $pr_number",
```

**Review by AntonReinhard — COMMENTED** [review]
Do we plan on having any automated tests for this functionality? I imagine it would be difficult because it would always rely on the internet and therefore would always be unstable (especially looking at the stability of github actions lately :eyes: )

**Comment by SimeonEhrig** [comment]
> Do we plan on having any automated tests for this functionality? I imagine it would be difficult because it would always rely on the internet and therefore would always be unstable (especially looking at the stability of github actions lately 👀 )

I have several json files, which I can add to test at least the json parser.

**Comment by AntonReinhard** [comment]
> I have several json files, which I can add to test at least the json parser.

Sounds good 👍

**Comment by SimeonEhrig** [comment]
After a discussion, we decided to not add tests. There was only one pretty easy test. The rest depends on the internet. Therefore, there would be a lot of work, to modify the test infrastructure of the integrationTests package.

Instead, I uploaded some response from the GitHub API for a PR targeting a dev branch, a PR targeting a main branch and an non existing branch.

**Pull request merged**